File: lab3/q5.cpp
// clang++ -std=c++17 -Wall -Werror -Wextra -Wpedantic -g3 -o linelength linelength.cpp
// cd r28chang/SYDE-121/assignment01

#include <iostream>
#include <string>
#include <cmath>
#include <vector>

using namespace std;

auto main() -> int
{
    std::string curr;
    double lines = 0;
    int rows = 0;
    int cols = 0;
    float currInt = 0.0;

    std::vector<float> values;
    std::vector<int> targets;

    while (getline(std::cin, curr))
    {
        lines++;
        currInt = stof(curr);

        if (lines == 1)
        {
            rows = stoi(curr);
        }
        else if (lines == 2)
        {
            cols = stoi(curr);
        }
        else if (lines > 2 && lines <= (rows * cols) +2)
        {
            values.push_back(currInt);
        }
        else {
            targets.push_back(stoi(curr)); 
        }
    }

    int counter = 0;
    vector<vector<float>> list(rows, vector<float>(cols, 0.0));

    for (int i = 0; i < rows; i++)
    {
        for (int j = 0; j < cols; j++)
        {
            list[i][j] = values[counter];
            counter++;
        }
    }

    // matrix header 
    std::cout << "Matrix\n-------------" << std::endl; 

    // print matrix
    for (int i = 0; i < rows; i++)
    {
        for (int j = 0; j < cols; j++)
        {
            std::cout << std::to_string(list[i][j]) + " ";
        }

        std::cout << "\n" << std::endl;
    }

    // targets header 
    std::cout << "Targets\n-------------" << std::endl; 
    
    int targRow = 0; 
    int targCol = 0; 
    int targPair = 1; 
    int targSize = targets.size(); 

    for (int i = 0; i < targSize ; i++){
        if (i % 2 == 0){
            targRow = targets[i]-1; 
        }
        else if (i % 2 != 0){
            targCol = targets[i]-1; 
            if (targRow >= rows || targCol >= cols){
                std::cout << "Target " + std::to_string(targPair) + ": ERROR - Target out of bounds" << std::endl; 
                targPair++; 
            }
            else {
                std::cout << "Target " + std::to_string(targPair) + ": " + std::to_string(list[targRow][targCol]) << std::endl; 
                targPair++; 
            }
        }
    }

    return 0;
}

File: lab3/q5.py
import sys 

values = []
targets = []
lines = 0 
rows = 0.0 
cols = 0.0 

for curr in sys.stdin: 
    lines += 1 
    curr_num = float(curr)

    if (lines == 1):
        rows = int(curr)
    elif (lines == 2):
        cols = int(curr)

    elif (lines > 2 and lines <= (rows * cols) + 2): 
        values.append(curr_num)

    else: 
        targets.append(int(curr))

counter = 0 

list = [[0 for x in range(cols)] for y in range(rows)] 

for i in range (0, rows, 1):
    for j in range(0, cols, 1):
        list[i][j] = values[counter]
        counter+=1  

print ("Matrix")
print ("-------------")

# print matrix 
for element in list:
    print(element)

print ("\nTargets")
print ("-------------")

targ_row = 0 
targ_col = 0 
targ_pair = 1 

for i in range (0, len(targets), 1):
    if (i % 2 == 0):
        targ_row = targets[i] - 1
    elif (i % 2 != 0):
        targ_col = targets[i] - 1 
        if (targ_row >= int(rows) or targ_col >= int(cols)):
            print("Target " + str(targ_pair) + ": ERROR - Target out of bounds")
            targ_pair += 1
        else: 
            print ("Target " + str(targ_pair) + ": " + str(list[targ_row][targ_col]))
            targ_pair += 1 

